pkg/server/server.go

```diff
@@ -306,11 +307,19 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		return nil, fmt.Errorf("request duration datastore count buckets must not be empty")
 	}
 
-	s.typesystemResolver = typesystem.MemoizedTypesystemResolverFunc(s.datastore)
+	s.typesystemResolver, s.typesystemResolverStop = typesystem.MemoizedTypesystemResolverFunc(s.datastore)
 
 	return s, nil
 }
 
+func (s *Server) Close() {
+	if s.checkCache != nil {
+		s.checkCache.Stop()
+	}
+	s.datastore.Close()
```

Comment:
Doesn't feel right to have the Server be the responsible party to close the datastore. The datastore is a dependency of the Server and is passed in. It should be the callers responsibility to close it because it is the callers responsibility to construct it. This approach leaks the responsibility of the caller vs callee.

Reply:
cc @willvedd @jpadilla @adriantam @sergiught for your opinions. Should the constructor of the datastore be the party responsible for closing it? If not, why do you feel the Server should be the responsible party?

Reply:
IMO, whoever create the object is responsible for closing it.  Otherwise, you will have a situation where the caller will still have reference to the object while the callee had close it.

Reply:
Oh! I forgot that the server doesn't actually instantiate the `datastore`, it merely takes a reference to it :) good point, i'll leave this one out

Reply:
I think its a well intentioned DX improvement but creator should be expected to close. Might be unnecessary risk otherwise.

I think an easy n+1 is to include the close in any public examples we have documented.